packages/ra-core/src/util/useCheckForApplicationUpdate.ts

```diff
@@ -28,18 +30,18 @@ export const useCheckForApplicationUpdate = (
     useEffect(() => {
         if (disabled) return;
 
-        getHashForUrl(url).then(hash => {
+        getHashForUrl(url, fetchOptions).then(hash => {
             if (hash != null) {
                 currentHash.current = hash;
             }
         });
-    }, [disabled, url]);
+    }, [disabled, url, fetchOptions]);
```

Comment:
I'm afraid that people will pass non-memoized objects as `fetchOptions`, which will cause too many rerenders. Perpaphs we should use the stringified version of the options in the dependency array?